sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -469,10 +476,51 @@ def analyze_property(
                 description=self.get_docstring(typ, name),
             )
         elif is_resource(arg):
-            # TODO: https://github.com/pulumi/pulumi/issues/18484
-            raise Exception(
-                f"Resource references are not supported yet: found type '{arg.__name__}' for '{typ.__name__}.{name}'"
-            )
+            type_string = getattr(arg, "pulumi_type", None)
+            if not type_string:
+                mod = arg.__module__.split(".")[0]
+                raise Exception(
+                    f"Can not determine resource reference for type '{arg.__name__}' used in '{typ.__name__}.{name}': "
+                    + f"'{arg.__name__}.pulumi_type' is not defined. This may be due to an outdated version of '{mod}'."
```

Comment:
Not sure what the best message here is. The codegen change will take a couple weeks to hit all the SDKs.

Reply:
Maybe `'{mod}' needs to depend on pulumi >= 3.1xx.0`.  Please contact the author to upgrade the SDK.`? But it's awkward because this depends on codegen, so there's no good way for the user to see which version something was codegene'd with :thinking: 

But not sure either.